docs/proposals/20210203-unmanaged-infrastructure.md

```diff
@@ -0,0 +1,235 @@
+
+There currently is no immutability support for CRD annotations within the Kubernetes API.
+
+This means that, once a user has created their externally managed InfraCluster, they could at some point, update the annotation to make the InfraCluster appear to be managed.
+
+There is no way to predict what would happen in this scenario.
+The InfraCluster controller would start attempting to reconcile infrastructure that it did not create, and therefore, there may be assumptions it makes that mean it cannot manage this infrastructure.
+
+To prevent this, we will have to implement (in the InfraCluster webhook) a means to prevent users converting externally managed InfraClusters into managed InfraClusters.
+
+Note however, converting from managed to externally managed should cause no issues and should be allowed.
+It will be documented as part of the externally managed contract that this is a one way operation.
+
+### Future Work
+
+#### Marking InfraCluster ready manually
+
+The content of this proposal assumes that the management of the external infrastructure is done by some controller which has the ability to set the spec and status of the InfraCluster resource.
+
+In reality, this may not be the case. For example, if the infrastructure was created by an admin using Terraform.
+
+When using a system such as this, a user can copy the details from the infrastructure into an InfraCluster resource and create this manually.
+However, they will not be able to set the InfraCluster to ready as this requires updating the resource status which is difficult when not using a controller.
+
+To allow users to adopt this external management pattern without the need for writing their own controllers or tooling, we will provide a longer term solution that allows a user to indicate that the infrastructure is ready and have the status set appropriately.
+
+The exact mechanism for how this will work is undecided, though the following ideas have been suggested:
+
+- Write a kubectl plugin that allows a user to mark their InfraCluster resources as ready
+
+- Add a secondary annotation to this contract that causes the provider InfraCluster controller to mark resources as ready
+
+## Alternatives
+
+### ExternalInfra CRD
+
+We could have an adhoc CRD https://github.com/kubernetes-sigs/cluster-api/issues/4095
+
+This would introduce complexity for the CAPI ecosystem with yet an additional CRD and it woudn't scale well across providers as it would need to contain provider specific information.
+
+### ManagementPolicy field
+
+As an alternative to the proposed annotation, a `ManagementPolicy` field on Infrastructure Cluster spec could be required as part of this contract.
+The field would be an enum that initially has 2 posisble values: managed and unmanaged.
+That would require a new provider contract and modification of existing infrastructure CRDs, so this option is not preferred.
```

Comment:
I think implementing it via an annotation is also a new provider contract. Nevertheless I also prefer implementing it via an annotation for now so we don't have to adjust all CRDs.